src/main/java/se/bjurr/violations/comments/gitlab/lib/GitLabCommentsProvider.java

```diff
@@ -20,184 +20,192 @@
+		final String targertSha = mergeRequest.getTargetSha();
+		final String newPath = file.getFilename();
+		final String oldPath = null;
+		try {
+			gitlabApi.createTextDiscussion(mergeRequest, discussionContent,
+					null, sourceSha, targertSha, headSha, newPath, newLine,
+					oldPath, oldLine);
+		} catch (final Throwable e) {
+			final String lineSeparator = System.lineSeparator();
+			LOGGER.error("Could not create diff discussion!" + lineSeparator +
+					"ProjectID: " + projectId + lineSeparator +
+					"SourceSha: " + sourceSha + lineSeparator +
+					"HeadSha: " + headSha + lineSeparator +
+					"TargetSha: " + targertSha + lineSeparator +
+					"Path " + newPath + lineSeparator +
+					"Violation: " + discussionContent + lineSeparator +
+					"NewLine " + newLine + ", OldLine" + newLine);
+		}
+	}
+
+	@Override
+	public List<Comment> getComments() {
+		final List<Comment> found = new ArrayList<>();
+		try {
+			final List<GitlabNote> notes = gitlabApi.getAllNotes(mergeRequest);
+			for (final GitlabNote note : notes) {
+				final String identifier = note.getId() + "";
+				final String content = note.getBody();
+				final String type = "PR";
+				final List<String> specifics = new ArrayList<>();
+				final Comment comment = new Comment(identifier, content, type, specifics);
+				found.add(comment);
+			}
+		} catch (final Throwable e) {
+			LOGGER.error("Could not get comments", e);
+		}
+		return found;
+	}
+
+	@Override
+	public List<ChangedFile> getFiles() {
+		final List<ChangedFile> changedFiles = new ArrayList<>();
+		for (final GitlabCommitDiff change : mergeRequest.getChanges()) {
+			final String filename = change.getNewPath();
+			final List<String> specifics = new ArrayList<>();
+			final String patchString = change.getDiff();
+			specifics.add(patchString);
+			final ChangedFile changedFile = new ChangedFile(filename, specifics);
+			changedFiles.add(changedFile);
+		}
+
+		return changedFiles;
+	}
+
+	@Override
+	public void removeComments(final List<Comment> comments) {
+		for (final Comment comment : comments) {
+			try {
+				final GitlabNote noteToDelete = new GitlabNote();
+				noteToDelete.setId(Integer.parseInt(comment.getIdentifier()));
+				gitlabApi.deleteNote(mergeRequest, noteToDelete);
+			} catch (final Throwable e) {
+				LOGGER.error("Could not delete note " + comment);
+			}
+		}
+	}
+
+	@Override
+	public boolean shouldComment(final ChangedFile changedFile, final Integer line) {
+		final String patchString = changedFile.getSpecifics().get(0);
+		final Optional<Integer> lineFoundOpt = findLineToComment(patchString, line);
+		final boolean commentOnlyChangedContent =
+				violationCommentsToGitLabApi.getCommentOnlyChangedContent();
+		return !commentOnlyChangedContent || lineFoundOpt.isPresent();
+	}
+
+	@Override
+	public boolean shouldCreateBulkComment() {
+		return violationCommentsToGitLabApi.getCreateCommentWithAllSingleFileComments();
+	}
+
+	@Override
+	public boolean shouldCommentOnTheDiff() {
```

Comment:
You probably did not mean to hard code this to false.

Reply:
Yes you are right. I still need to add a config parameter for that one. I will do it as soon as possible.